code/ATMOSPHERICS/hvac/spaceheater.dm

```diff
@@ -38,25 +68,38 @@
 		overlays  += "[base_state]-open"
 	return
 
+/obj/machinery/space_heater/campfire/update_icon()
+	overlays.len = 0
+	if(on)
+		var/fireintensity = min(Floor((cell.charge-1)/(cell.maxcharge/4))+1,4)
```

Comment:
This can be abused in all sorts of strange ways (for example, EMP). Just make it use a special var for fire fuel instead and can the battery